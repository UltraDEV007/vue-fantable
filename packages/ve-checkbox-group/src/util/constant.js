// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
