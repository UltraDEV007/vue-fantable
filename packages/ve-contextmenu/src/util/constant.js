// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
