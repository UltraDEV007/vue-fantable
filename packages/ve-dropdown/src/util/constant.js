// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
