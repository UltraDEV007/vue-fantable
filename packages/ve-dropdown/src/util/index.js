/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
