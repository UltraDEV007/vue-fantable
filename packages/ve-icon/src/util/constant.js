// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
