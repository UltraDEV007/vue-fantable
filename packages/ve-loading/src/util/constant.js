// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
