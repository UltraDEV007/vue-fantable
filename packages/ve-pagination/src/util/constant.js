// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
