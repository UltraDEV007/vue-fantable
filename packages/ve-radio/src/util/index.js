import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
