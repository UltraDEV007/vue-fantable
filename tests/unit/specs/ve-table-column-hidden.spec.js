import { mount } from "@vue/test-utils";
import { later } from "../util";
import veTable from "@/ve-table";

describe("veTable column hidden", () => {
    const TABLE_DATA = [
        {
            col1: "col1-1",
            col2: "col2-1",
            col3: "col3-1",
            col4: "col4-1",
            col5: "col5-1",
            col6: "col6-1",
            col7: "col7-1",
            col8: "col8-1",
        },
    ];

    const COLUMNS = [
        { field: "col1", key: "col1", title: "col1", width: "10%" },
        {
            title: "col2-col3",
            children: [
                {
                    field: "col2",
                    key: "col2",
                    title: "col2",
                    width: 100,
                },
                {
                    field: "col3",
                    key: "col3",
                    title: "col3",
                    width: 110,
                },
            ],
        },
        {
            title: "col4-col5-col6",
            children: [
                {
                    title: "col4-col5",
                    children: [
                        {
                            field: "col4",
                            key: "col4",
                            title: "col4",
                            width: 130,
                        },
                        {
                            field: "col5",
                            key: "col5",
                            title: "col5",
                            width: 140,
                        },
                    ],
                },
                {
                    title: "col6",
                    field: "col6",
                    key: "col6",
                    width: 140,
                },
            ],
        },
        { field: "col7", key: "col7", title: "col7", width: 150 },
        { field: "col8", key: "col8", title: "col8", width: 160 },
    ];

    it("columnHiddenOption defaultHiddenColumnKeys prop snapshot", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                columnHiddenOption: {
                    // default hidden column keys
                    defaultHiddenColumnKeys: ["col8"],
                },
            },
        });

        expect(wrapper.vm.hiddenColumns).toEqual(["col8"]);

        const firstHeaderTr = wrapper.findAll(
            ".ve-table-header .ve-table-header-tr",
        );

        const Ths = firstHeaderTr.at(0).findAll(".ve-table-header-th");

        expect(Ths.at(Ths.length - 1).text()).toEqual("col7");
    });

    it("columnHiddenOption defaultHiddenColumnKeys prop", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                columnHiddenOption: {
                    // default hidden column keys
                    defaultHiddenColumnKeys: ["col1", "col2", "col3"],
                },
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("instance method hideColumnsByKeys", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                columnHiddenOption: {
                    // default hidden column keys
                    defaultHiddenColumnKeys: ["col8"],
                },
            },
        });

        wrapper.vm.hideColumnsByKeys(["col1", "col2"]);

        expect(wrapper.vm.hiddenColumns).toEqual(["col8", "col1", "col2"]);

        await later();

        expect(wrapper.vm.groupColumns).toEqual([
            [
                {
                    title: "col2-col3",
                    children: [
                        {
                            field: "col3",
                            key: "col3",
                            title: "col3",
                            width: 110,
                            fixed: undefined,
                            _level: 2,
                            _keys: "col3",
                            _colspan: 1,
                            _rowspan: 2,
                        },
                    ],
                    _level: 1,
                    _keys: "col3|",
                    _colspan: 1,
                    _rowspan: 1,
                },
                {
                    title: "col4-col5-col6",
                    children: [
                        {
                            title: "col4-col5",
                            children: [
                                {
                                    field: "col4",
                                    key: "col4",
                                    title: "col4",
                                    width: 130,
                                    fixed: undefined,
                                    _level: 3,
                                    _keys: "col4",
                                    _colspan: 1,
                                    _rowspan: 1,
                                },
                                {
                                    field: "col5",
                                    key: "col5",
                                    title: "col5",
                                    width: 140,
                                    fixed: undefined,
                                    _level: 3,
                                    _keys: "col5",
                                    _colspan: 1,
                                    _rowspan: 1,
                                },
                            ],
                            fixed: undefined,
                            _level: 2,
                            _keys: "col4|col5|",
                            _colspan: 2,
                            _rowspan: 1,
                        },
                        {
                            title: "col6",
                            field: "col6",
                            key: "col6",
                            width: 140,
                            fixed: undefined,
                            _level: 2,
                            _keys: "col6",
                            _colspan: 1,
                            _rowspan: 2,
                        },
                    ],
                    _level: 1,
                    _keys: "col4|col5|col6|",
                    _colspan: 3,
                    _rowspan: 1,
                },
                {
                    field: "col7",
                    key: "col7",
                    title: "col7",
                    width: 150,
                    _level: 1,
                    _keys: "col7",
                    _colspan: 1,
                    _rowspan: 3,
                },
            ],
            [
                {
                    field: "col3",
                    key: "col3",
                    title: "col3",
                    width: 110,
                    fixed: undefined,
                    _level: 2,
                    _keys: "col3",
                    _colspan: 1,
                    _rowspan: 2,
                },
                {
                    title: "col4-col5",
                    children: [
                        {
                            field: "col4",
                            key: "col4",
                            title: "col4",
                            width: 130,
                            fixed: undefined,
                            _level: 3,
                            _keys: "col4",
                            _colspan: 1,
                            _rowspan: 1,
                        },
                        {
                            field: "col5",
                            key: "col5",
                            title: "col5",
                            width: 140,
                            fixed: undefined,
                            _level: 3,
                            _keys: "col5",
                            _colspan: 1,
                            _rowspan: 1,
                        },
                    ],
                    fixed: undefined,
                    _level: 2,
                    _keys: "col4|col5|",
                    _colspan: 2,
                    _rowspan: 1,
                },
                {
                    title: "col6",
                    field: "col6",
                    key: "col6",
                    width: 140,
                    fixed: undefined,
                    _level: 2,
                    _keys: "col6",
                    _colspan: 1,
                    _rowspan: 2,
                },
            ],
            [
                {
                    field: "col4",
                    key: "col4",
                    title: "col4",
                    width: 130,
                    fixed: undefined,
                    _level: 3,
                    _keys: "col4",
                    _colspan: 1,
                    _rowspan: 1,
                },
                {
                    field: "col5",
                    key: "col5",
                    title: "col5",
                    width: 140,
                    fixed: undefined,
                    _level: 3,
                    _keys: "col5",
                    _colspan: 1,
                    _rowspan: 1,
                },
            ],
        ]);
    });

    it("instance method showColumnsByKeys", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                columnHiddenOption: {
                    // default hidden column keys
                    defaultHiddenColumnKeys: ["col8", "col1", "col2"],
                },
            },
        });

        wrapper.vm.showColumnsByKeys(["col1", "col2"]);

        expect(wrapper.vm.hiddenColumns).toEqual(["col8"]);

        await later();

        expect(wrapper.vm.groupColumns).toEqual([
            [
                {
                    field: "col1",
                    key: "col1",
                    title: "col1",
                    width: "10%",
                    _level: 1,
                    _keys: "col1",
                    _colspan: 1,
                    _rowspan: 3,
                },
                {
                    title: "col2-col3",
                    children: [
                        {
                            field: "col2",
                            key: "col2",
                            title: "col2",
                            width: 100,
                            fixed: undefined,
                            _level: 2,
                            _keys: "col2",
                            _colspan: 1,
                            _rowspan: 2,
                        },
                        {
                            field: "col3",
                            key: "col3",
                            title: "col3",
                            width: 110,
                            fixed: undefined,
                            _level: 2,
                            _keys: "col3",
                            _colspan: 1,
                            _rowspan: 2,
                        },
                    ],
                    _level: 1,
                    _keys: "col2|col3|",
                    _colspan: 2,
                    _rowspan: 1,
                },
                {
                    title: "col4-col5-col6",
                    children: [
                        {
                            title: "col4-col5",
                            children: [
                                {
                                    field: "col4",
                                    key: "col4",
                                    title: "col4",
                                    width: 130,
                                    fixed: undefined,
                                    _level: 3,
                                    _keys: "col4",
                                    _colspan: 1,
                                    _rowspan: 1,
                                },
                                {
                                    field: "col5",
                                    key: "col5",
                                    title: "col5",
                                    width: 140,
                                    fixed: undefined,
                                    _level: 3,
                                    _keys: "col5",
                                    _colspan: 1,
                                    _rowspan: 1,
                                },
                            ],
                            fixed: undefined,
                            _level: 2,
                            _keys: "col4|col5|",
                            _colspan: 2,
                            _rowspan: 1,
                        },
                        {
                            title: "col6",
                            field: "col6",
                            key: "col6",
                            width: 140,
                            fixed: undefined,
                            _level: 2,
                            _keys: "col6",
                            _colspan: 1,
                            _rowspan: 2,
                        },
                    ],
                    _level: 1,
                    _keys: "col4|col5|col6|",
                    _colspan: 3,
                    _rowspan: 1,
                },
                {
                    field: "col7",
                    key: "col7",
                    title: "col7",
                    width: 150,
                    _level: 1,
                    _keys: "col7",
                    _colspan: 1,
                    _rowspan: 3,
                },
            ],
            [
                {
                    field: "col2",
                    key: "col2",
                    title: "col2",
                    width: 100,
                    fixed: undefined,
                    _level: 2,
                    _keys: "col2",
                    _colspan: 1,
                    _rowspan: 2,
                },
                {
                    field: "col3",
                    key: "col3",
                    title: "col3",
                    width: 110,
                    fixed: undefined,
                    _level: 2,
                    _keys: "col3",
                    _colspan: 1,
                    _rowspan: 2,
                },
                {
                    title: "col4-col5",
                    children: [
                        {
                            field: "col4",
                            key: "col4",
                            title: "col4",
                            width: 130,
                            fixed: undefined,
                            _level: 3,
                            _keys: "col4",
                            _colspan: 1,
                            _rowspan: 1,
                        },
                        {
                            field: "col5",
                            key: "col5",
                            title: "col5",
                            width: 140,
                            fixed: undefined,
                            _level: 3,
                            _keys: "col5",
                            _colspan: 1,
                            _rowspan: 1,
                        },
                    ],
                    fixed: undefined,
                    _level: 2,
                    _keys: "col4|col5|",
                    _colspan: 2,
                    _rowspan: 1,
                },
                {
                    title: "col6",
                    field: "col6",
                    key: "col6",
                    width: 140,
                    fixed: undefined,
                    _level: 2,
                    _keys: "col6",
                    _colspan: 1,
                    _rowspan: 2,
                },
            ],
            [
                {
                    field: "col4",
                    key: "col4",
                    title: "col4",
                    width: 130,
                    fixed: undefined,
                    _level: 3,
                    _keys: "col4",
                    _colspan: 1,
                    _rowspan: 1,
                },
                {
                    field: "col5",
                    key: "col5",
                    title: "col5",
                    width: 140,
                    fixed: undefined,
                    _level: 3,
                    _keys: "col5",
                    _colspan: 1,
                    _rowspan: 1,
                },
            ],
        ]);
    });
});
