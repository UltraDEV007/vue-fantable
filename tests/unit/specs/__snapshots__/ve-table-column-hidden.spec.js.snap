// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable column hidden columnHiddenOption defaultHiddenColumnKeys prop 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 130px;">
            <col style="width: 140px;">
            <col style="width: 140px;">
            <col style="width: 150px;">
            <col style="width: 160px;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="3" class="fan-table-header-th" style="text-align: center; top: 0px;">col4-col5-col6</th>
              <th rowspan="3" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col7</th>
              <th rowspan="3" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: center; top: 0px;">col8</th>
            </tr>
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="2" class="fan-table-header-th" style="text-align: center; top: 0px;">col4-col5</th>
              <th rowspan="2" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col6</th>
            </tr>
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col4</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col5</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="col4" class="fan-table-body-td" style="text-align: center;">col4-1</td>
              <td rowspan="1" colspan="1" col-key="col5" class="fan-table-body-td" style="text-align: center;">col5-1</td>
              <td rowspan="1" colspan="1" col-key="col6" class="fan-table-body-td" style="text-align: center;">col6-1</td>
              <td rowspan="1" colspan="1" col-key="col7" class="fan-table-body-td" style="text-align: center;">col7-1</td>
              <td rowspan="1" colspan="1" col-key="col8" class="fan-table-body-td" style="text-align: center;">col8-1</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;
